js/modules.js

```diff
@@ -15,52 +15,79 @@ CORE.create_module("email-nav", function (sb) {
 CORE.create_module("email-catalog", function (sb) {
     var emails;
 
+    var openEmail = function(e) {
+        sb.hide();
+        sb.notify({
+            type : "open-email",
+            data : e.currentTarget.getAttribute("data-email-id")
```

Comment:
There would be a separate "table" of data for each user right, so it would be only querying emails that would be for them. So I guess ID's could overlap for different users, but for a single user, each ID is unique